crates/ruff_linter/src/rules/pylint/rules/iteration_over_set.rs

```diff
@@ -46,8 +47,21 @@ pub(crate) fn iteration_over_set(checker: &mut Checker, expr: &Expr) {
         return;
     };
 
-    if set.iter().any(Expr::is_starred_expr) {
-        return;
+    let mut seen_values: FxHashSet<ComparableExpr> = FxHashSet::default();
+
+    for value in set {
+        if value.is_literal_expr() {
+            let comparable_value = ComparableExpr::from(value);
+
+            if !seen_values.insert(comparable_value) {
+                // if the set contains a duplicate literal value, early exit.
+                // rule `B033` can catch that.
+                return;
+            }
+        } else {
+            // If the set contains a non-literal expression, early exit.
+            return;
+        }
```

Comment:
Nit: We can avoid building the hash set and then exiting after we added a few values because there's a non-literal by first testing if all entries are literals

```suggestion
		if set.iter().any(|value| !value.is_literal_expr()) {
			return;
		}
		
		let mut seen_values = FxHashSet::with_capacity(set.len());

    for value in set {
        let comparable_value = ComparableExpr::from(value);
        if !seen_values.insert(comparable_value) {
            // if the set contains a duplicate literal value, early exit.
            // rule `B033` can catch that.
            return;
        }
     }
```

Reply:
Good call. Implemented!